core/sql/common/DateTimeType.cpp

```diff
@@ -1399,18 +1403,69 @@ NABoolean SQLMPDatetime::isCompatible(const NAType& other, UInt32 * flags) const
 //  DatetimeValue : A datetime value
 //
 // ***********************************************************************
-
 DatetimeValue::DatetimeValue
 ( const char* strValue
 , rec_datetime_field startField
 , rec_datetime_field endField
 , UInt32& fractionPrecision   /* IN - OUT */
+, NABoolean useOldConstructor
 )
 : value_(NULL)
 , valueLen_(0)
 , startField_(startField)
 , endField_  (endField)
 {
+  if (useOldConstructor)
+    {
+      oldConstructor(strValue, startField, endField, fractionPrecision);
+      return;
+    }
+
+  char dstValue[101];
+
+  ComDiagsArea *diags = NULL;
+
+  Lng32 trueFP = 6;
+  ULng32 flags = 0;
+  flags |= CONV_NO_HADOOP_DATE_FIX;
+  short rc = 
+    ExpDatetime::convAsciiToDatetime((char*)strValue, strlen(strValue), 
+                                     dstValue, 100,
+                                     startField, endField, 
+                                     ExpDatetime::DATETIME_FORMAT_NONE,
+                                     trueFP,
+                                     NULL, &diags, flags);
+  if (rc)
+    return;
```

Comment:
That return is for error paths. The Datetime value being constructed will be invalid and
not be used. isValid() method will return true.
But we can initialize fractionPrecision to zero on entry to this method.